### PR TITLE
Part 1: Add validation when amount to withdraw is higher than the balance for the address.

### PR DESCRIPTION
**Pull request by aguschirico**
This PR adds the require to make sure amount to withdraw doesn't overpass the balance the specified token for that address.
 - Added solidity waffle expect extensions
 - Added test to make sure the swap reverts when transfer amount exceeds allowance (ERC777)
 - Added tests for the validation of the current balance on withdraw operations
 - Upgrades chai npm package
 This PR is the first part of a series of 2. The second part will add functionality related how the ERC777 hooks are handled.

### CODE REVIEW

File: test/Swap.test.ts [hunk 1]
```diff
@@ -29,21 +33,47 @@ describe("Swap RBTC", function () {
...
+      to: swapRBTC.address,
+      value: oneEther
+    });
+
+    await expect(swapRBTC.connect(sender).swapWRBTCtoRBTC(halfEther, sideTokenBtc.address))
+      .to.be.revertedWith("ERC777: transfer amount exceeds allowance");
+  });
+
+  it("Should Not be allowed to withdraw RBTC When balance is not enough", async function () {
+    await sideTokenBtc.connect(minter).mint(sender.address, halfEther, "0x", "0x");
```
Comment (pmprete) [inline test/Swap.test.ts:71]: I think it's not necesary to mint tokens to the sender for this test

File: test/Swap.test.ts [hunk 2]
```diff
@@ -29,21 +33,47 @@ describe("Swap RBTC", function () {
...
+    await expect(swapRBTC.connect(sender).withdrawalRBTC(halfEther)).to.be.revertedWith("SwapRBTC: amount > senderBalance");
+  });
+
+  it("Should Not be allowed to withdraw WRBTC When balance is not enough", async function () {
+    await sideTokenBtc.connect(minter).mint(sender.address, halfEther, "0x", "0x");
```
Comment (pmprete) [inline test/Swap.test.ts:76]: Same here

**Review by pmprete — APPROVED** [review]
Except my comment looks good, nice job 👍

File: test/Swap.test.ts (hunk 1 above)
Comment (aguschirico) [reply test/Swap.test.ts:71]: You're right. Fixed.

File: test/Swap.test.ts (hunk 2 above)
Comment (aguschirico) [reply test/Swap.test.ts:76]: Fixed.

**Review by pmprete — APPROVED** [review]
Great work!

**Pull request merged**